src/Agent.Sdk/Util/PlatformUtil.cs

```diff
@@ -355,6 +358,59 @@ public async static Task<bool> DoesSystemPersistsInNet6Whitelist()
 
             return net6SupportedSystems.Any((s) => s.Equals(systemId));
         }
+        public static bool DetectDockerContainer()
+        {
+            bool isDockerContainer = false;
+
+            try
+            {
+                if (PlatformUtil.RunningOnWindows)
+                {
+                    // For Windows we check Container Execution Agent Service (cexecsvc) existence
+                    var serviceName = "cexecsvc";
+                    ServiceController[] scServices = ServiceController.GetServices();
+                    if (scServices.Any(x => String.Equals(x.ServiceName, serviceName, StringComparison.OrdinalIgnoreCase) && x.Status == ServiceControllerStatus.Running))
+                    {
+                        isDockerContainer = true;
+                    }
+                }
+                else
+                {
+                    // In Unix in control group v1, we can identify if a process is running in a Docker
+                    var initProcessCgroup = File.ReadLines("/proc/1/cgroup");
+                    if (initProcessCgroup.Any(x => x.IndexOf(":/docker/", StringComparison.OrdinalIgnoreCase) >= 0))
+                    {
+                        isDockerContainer = true;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                // Logging exception will be handled by JobRunner
+                throw;
+            }
```

Comment:
Very low priority: try {} catch {} doesn't have any impact since the exception is just rethrown and "throw;" preserves the callstack.   My suggestion is just remove the try / catch